File: cuarray/samples/cpp/src/CuArray_m.cpp
#include <cuarray.h>
#include <iostream>

int main() {
    std::cout
            << "Running "
            << __FILE__
            << std::endl;
/* Creates a new float CuArray instance */
    CuArray<float> *cuArray = new CuArray<float>();

/*
 * Initializes the CuArray with 10 rows and 5 rows
 * and allocates memory on host.
 */
    cuArray->init(10,
                  5);

/* Get the number of rows in the CuArray */
    int m = cuArray->m();

/* Print the number of rows */
    std::cout
            << "Number of rows: "
            << m
            << std::endl;
/* Output:
 * Number of rows: 10
 */

    delete cuArray;
    return 0;
}

File: cuarray/samples/python/CuArray_m.py
import numpy as np
"""
Always precede CuArray with the data type
Here we are importing float template.
"""
from cuarray import FloatCuArray

print("Running", __file__)


"""Create a new float CuArray instance"""
float_cuarray = FloatCuArray()

"""Initialize the float CuArray with 10 rows and 2 columns"""
float_cuarray.init(10, 2)

"""Print the number of rows in the CuArray"""
print(float_cuarray.m())